lib/spack/spack/test/llnl/util/file_list.py

```diff
@@ -13,6 +14,10 @@
 
 import spack.paths
 
+if sys.platform == "win32":
+    from pathlib import WindowsPath
+else:
+    from pathlib import PosixPath
```

Comment:
I'm aware there remain a lot of bad examples across the repo, but one of the most useful improvements we could make regarding windows support is to remove as many of these `if sys.platform == "win32"` checks as possible. Preferably, we would take the approach of #21720, where any functionality that diverges across platforms or python versions would be exposed in `lib/spack/llnl/util/compat.py` under the same name.

Reply:
However, it also looks like neither `WindowsPath` or `PosixPath` are used in this file? I suspect `spack style` will have failed; please ensure that it passes before you request review in the future. Please feel free to ask questions here or over slack if it's not clear how to resolve a style error.

Reply:
Ah, when I try to run `spack style` I see that it fails to even execute because of a runtime type error:
```bash
$ spack style
==> Error: 'PosixPath' object has no attribute 'endswith'
```

When making very broad changes across files used all across the repo (such as those in `llnl.util`), I strongly recommend trying to run `spack style` at the start, adding mypy types as you go, and then running `spack style` and `spack unit-test -k filesystem.py` every time you change something so you can immediately track down whenever your change breaks something else.

Reply:
Ugh, this was actually passing the CI before I made changes this morning (and did not test again... I know, not the best), I'll fix all that up, I was more concerned with getting eyes on the changes/some buyin

Reply:
I appreciate it and I'm glad to hear you had it working recently!! No rush at all, thanks for clarifying!

Reply:
With regard to the if statement conditional on platforms, I can't explain why that's there... It should be removed entirely, possible a rebase mistake I made.

Reply:
Oh ok great! Then we can resolve this comment!